docs/spec_draft.md

```diff
@@ -941,6 +942,69 @@ produces an implementation-defined value.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.dynamic_update_slice
+
+### Semantics
+
+Overwrites the `operand` tensor at `start_indices` with values from `update`
+tensor and produces a `result` tensor.
+
+More formally, `result[j0, ..., jR-1]` is defined as:
+  * `update[i0, ..., iR-1]` if `jd = effective_start_indices[d] + id` where
+    `effective_start_indices[d] = clamp(start_indices[d], 0, dim(operand, d) - `
+    `dim(update, d))`
+  * `result[i0, ..., iR-1] = operand[i0, ..., iR-1]` otherwise.
+
+### Inputs
+
+| Name            | Type                                                    |
+|-----------------|---------------------------------------------------------|
+| `operand`       | tensor of any supported type                            |
+| `update`        | tensor of any supported type                            |
+| `start_indices` | variadic number of 0-dimensional tensors of type `si64` |
+
+### Outputs
+
+| Name     | Type                         |
+|----------|------------------------------|
+| `result` | tensor of any supported type |
+
+### Constraints
+
+  * (C1) `operand` and `result` have the same type.
+  * (C2) element_type(`update`) $=$ element_type(`operand`).
+  * (C3) rank(`update`) $=$ rank(`operand`).
+  * (C4) size(`start_indices`) $=$ rank(`operand`).
+  * (C5) dim(`update`, `k`) $\in$ [0, dim(`operand`, `k`)) for all `k` $\in$
+    [0, rank(`operand`)).
+
+### Examples
+
+```mlir
+// %operand: [
+//            [1, 1, 1, 1],
+//            [1, 1, 0, 0],
+//            [1, 1, 0, 0],
+//            [1, 1, 1, 1]
+//           ]
+// %update: [
+//           [1, 1],
+//           [1, 1]
+//          ]
+// %start_indices0: 1
+// %start_indices1: 2
```

Comment:
Let's change start indices similarly to https://github.com/openxla/stablehlo/pull/556#discussion_r1030773820.